e2e-tests/spring-rest-openapi-v2/src/test/java/org/evomaster/e2etests/spring/examples/displayissue/DisplayIssueEMTest.java

```diff
@@ -0,0 +1,110 @@
+    and the statistics match. After that, we run a blackbox test using its swagger, check the output and the generated
+    statistics file.
+     */
+    @Test
+    public void testRunBlackboxAndWhiteBox() throws Throwable {
+
+        // redirect console output to PrintStream
+
+        // old print stream
+        PrintStream old = System.out;
+
+        try {
+            runTestHandlingFlakyAndCompilation(
+                    "DisplayIssueEM",
+                    "org.DisplayIssueEM",
+                    5,
+                    (args) -> {
+
+                        ByteArrayOutputStream byteArrayOutput = new ByteArrayOutputStream();
+                        System.setOut(new PrintStream(byteArrayOutput));
+
+                        args.add("--writeStatistics");
+                        args.add("true");
+
+                        Solution<RestIndividual> solutionWhite = initAndRun(args);
+
+                        String expectedNumberOfPotentialFaultsWhite = findValueOfItemWithKeyInStats
+                                (solutionWhite, "potentialFaults");
+                        String expectedNumberOfCoveredTargetsWhite = findValueOfItemWithKeyInStats
+                                (solutionWhite, "coveredTargets");
+
+                        Assertions.assertTrue(byteArrayOutput.toString().contains("Covered targets " +
+                                "(lines, branches, faults, etc.): " + expectedNumberOfCoveredTargetsWhite));
+
+                        Assertions.assertTrue(byteArrayOutput.toString().contains("Potential faults: "
+                                + expectedNumberOfPotentialFaultsWhite));
+
+                        // now run the same thins as a blackbox test
+                        byteArrayOutput = new ByteArrayOutputStream();
+                        System.setOut(new PrintStream(byteArrayOutput));
+
+                        args.add("--blackBox");
+                        args.add("true");
+                        args.add("--bbTargetUrl");
+                        args.add(baseUrlOfSut);
+                        args.add("--bbSwaggerUrl");
+                        args.add(baseUrlOfSut + "/v2/api-docs");
+                        args.add("--outputFormat");
+                        args.add("JAVA_JUNIT_4");
+
+                        Solution<RestIndividual> solutionBlack = initAndRun(args);
+
+                        String expectedNumberOfPotentialFaultsBlack = findValueOfItemWithKeyInStats
+                                (solutionBlack, "potentialFaults");
+
+                        Assertions.assertTrue(byteArrayOutput.toString().contains("Potential faults: "
+                                + expectedNumberOfPotentialFaultsBlack));
+
+                    });
+        }
+        catch (Exception e) {
+            System.err.println("Exception occurred during the test case: ");
+        }
+        finally {
+
+            File statsFile = new File("statistics.csv");
```

Comment:
what if the name `statistics.csv` in `EMConfig` will change in future?  rather put it in a variable, and pass it as arguments with `statisticsFile` to make sure pointing to same file

Reply:
Hi Andrea,

Thank you so much for your comment. I have made that filename as a constant.